assets/js/api.js

```diff
@@ -1,644 +1,368 @@
+      } else {
+        crimesDb.insert(crimesTaffyList);
+      }
+      crimeCount += crimesInNbhood;
+      //$('#crimetotal').append(crimesInNbhood + '<br>');
 
-  function zoomToNeighborhood(nbhoodVal) {
-    timelineLayer.clearLayers();
-    if (nbhoodVal == "all") {
-      nbhoodLayer.clearLayers();
-      map.fitBounds(pdxBounds);
-    } else {
-      var hoodBbxArray = nbhoodDb({name: nbhoodVal}).first().bbx;
-      currentHoodBbx = (hoodBbxArray[0].concat(hoodBbxArray[1])).join(',');
-      hoodBbxArray[0] = switchCoords(hoodBbxArray[0]);
-      hoodBbxArray[1] = switchCoords(hoodBbxArray[1]);
-      map.fitBounds(hoodBbxArray, {
-        padding: [60, 90]
-      });
-      hoodBbxArray[0] = switchCoords(hoodBbxArray[0]);
-      hoodBbxArray[1] = switchCoords(hoodBbxArray[1]);
-      getNbhoodShape(nbhoodVal);
-    }
+    })
+    .fail(function() {
+      console.log("Failed to fetch crimes.");
+    });
+    crimesPromises.push(promise);
   }
 
-  function getPermitColor(year) {
-    return year >= 2015 ? '#E65100' :
-           year > 2014 ? '#DB4D00' :
-           year > 2013 ? '#EF6C00' :
-           year > 2012 ? '#FF7300' :
-           year > 2011 ? '#F57C00' :
-           year > 2010 ? '#FF850A' :
-           year > 2009 ? '#FB8C00' :
-           year > 2008 ? '#FF930F' :
-           year > 2007 ? '#FF9800' :
-           year > 2006 ? '#FFA114' :
-           year > 2005 ? '#FFA726' :
-           year > 2004 ? '#FFAD33' :
-           year > 2003 ? '#FFB442' :
-           year > 2002 ? '#FFBA52' :
-           year > 2001 ? '#FFC061' :
-           year > 2000 ? '#FFC46B' :
-           year > 1999 ? '#FFC875' :
-           year > 1998 ? '#FFCC80' :
-           year > 1997 ? '#FFD08A' :
-           year > 1996 ? '#FFD494' :
-           year >= 1995 ? '#FFD89E':
-                       '#FFD89E';
-  }
+  $.when.apply($, crimesPromises).done(function() {
+    $('#crimetotal').append(crimeCount);
+    nbhoodLayer.setStyle({
+      fillColor: getColor(crimeCount)
+    });
+  }).fail(function() {
+    console.log('Failed to finish iterating over crime.');
+  });
 
-  function getDemolitionsColor(year) {
-    return year >= 2015 ? '#4A148C' :
-           year > 2014 ? '#6A1B9A' :
-           year > 2013 ? '#7B1FA2' :
-           year > 2012 ? '#8E24AA' :
-           year > 2011 ? '#9C27B0' :
-           year > 2010 ? '#AB47BC' :
-           year > 2009 ? '#BA68C8' :
-           year > 2008 ? '#CE93D8' :
-           year > 2007 ? '#E1BEE7' :
-           year > 2006 ? '#E9CFED' :
-           year > 2005 ? '#EEDAF1' :
-           year >= 2004 ? '#F3E5F5' :
-                       '#F3E5F5';
+}
+
+// function getTotalCrimes(nbhood, yearRange) {
+//   var crimeCount = 0;
+//   for (var i = 0; i < yearRange.length; i++) {
+//     getCrimesYear(nbhood, yearRange[i]);
+//     crimeCount += getCrimesYear(nbhood, yearRange[i]);
+//   }
+//   return crimeCount;
+// }
+
+// Bind or update dom elements once they're loaded
```

Comment:
I didn't touch any of the code related to crimes, even the commented out parts, since I'm not sure how we are going to proceed with the crimes layer and the refactoring of the api calls in general. It seems to make sense to move the event handlers so that api.js only contains code that makes the ajax calls. We can do that in the second round of refactoring.